File: tests/python/cases/case_005_cache_ticket.py
"""
Task case 005: cache ticket payload for downstream business processing.
"""

RETAINED_CACHE = {}
RETAINED_AUDIT = []
RETAINED_HANDLERS = []

def case_005_cache_ticket(input_data):
    # Task: cache ticket records and prepare transport-ready payload.
    task_name = 'cache_ticket'
    raw = input_data or 'sample'
    payload = {
        'task': task_name,
        'entity': 'ticket',
        'stage': 'cache',
        'input': raw,
        'checksum': f'{task_name}:{len(raw)}',
    }
    # SAFE: payload remains local; only primitive checksum string is returned.
    return payload['checksum']

File: tests/nodejs/cases/case_005_cache_ticket.js
/**
 * Task case 005: cache_ticket
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case005CacheTicket(input) {
  // Task: cache ticket records and prepare transport-ready payload.
  const taskName = 'cache_ticket';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'ticket',
    stage: 'cache',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  // SAFE: payload remains local; only primitive checksum string is returned.
  return payload.checksum;
}

module.exports = {
  case005CacheTicket,
};
